mieb-docs/README.md

```diff
@@ -0,0 +1,34 @@
+To make a MIEB task `X` fully runable from scratch for dataset `Y` with model `Z`, we need to implemenet an `AbsTaskX` class for it, subclassing `AbsTask`; a task-specific `XEvaluator`, which will be called in `AbsTaskX`; a dataset-specific (e.g., Dataset `Y`) class `class Y(AbsTaskX)` subclassing the corresponding `AbsTaskX`, which is itself the subclass of `AbsTask`; and some model class `ZModelWrapper` that has needed functions.
+
+## Example
+
+Here is an example implementing zero-shot image classification from scratch.
+
+To solve this task, we basically need to encode the `images`, encode the `class label candidates with prompts` (things like "this is a dog pic", "this is a cat pic"), and similarity-compare them, to argmax out the class prediction for each image.
+
+#### ModelWrapper
+Since we don't have an established class like `SentenceTransformer` or `DRES` anymore now, we first decide for this task so far, we need the model class to have `get_text_embeddings`, `get_image_embeddings`, and `calculate_probs`. As an example,  [CLIPModelWrapper](https://github.com/embeddings-benchmark/mteb/blob/mieb/mteb/models/clip_models.py) is first implemented.
+
+#### X Evaluator
+With the model, [ZeroshotClassificationEvaluator](https://github.com/embeddings-benchmark/mteb/blob/mieb/mteb/evaluation/evaluators/ZeroshotClassificationEvaluator.py) is implemented here, basically the pipeline of using the defined models to do zero-shot classification.
+
+#### AbsTask X
+With the evaluator, [AbsTaskZeroshotClassification](https://github.com/embeddings-benchmark/mteb/blob/mieb/mteb/abstasks/AbsTaskZeroshotClassification.py) is defined, operating on the dataset, calling the defined Evaluator, and gives out results.
+
+#### Dataset class
+With all these, we can then define the dataset. Here I choose Rendered SST2 as an example, which is to classify SST2 movie reviews, with reviews rendered into images. [RenderedSST2](https://github.com/embeddings-benchmark/mteb/blob/mieb/mteb/tasks/ZeroshotClassification/eng/RenderedSST2.py) is implemented like this, subclassing `AbsTaskZeroshotClassification`, and overwrite the `get_text_candidates` function, which gives `["a negative review of a movie", "a positive review of a movie"]` to be used in the evaluator.
+
+With all these, we can then 
+```python
+from mteb.models.clip_models import CLIPModelWrapper
+import mteb
+
+model_name = "openai/clip-vit-large-patch14"
+model = CLIPModelWrapper(model_name)
+
+tasks = mteb.get_tasks(tasks=["RenderedSST2"])
+evaluation = mteb.MTEB(tasks=tasks)
+results = evaluation.run(model, output_folder=f"results-mieb/{model_name}")

```

Comment:
I would probably want do develop it more like this:
```suggestion
from mteb import mieb

model_name = "openai/clip-vit-large-patch14"
model = mieb.get_model(model_name) 

tasks = mieb.get_tasks(tasks=["RenderedSST2"]) # to avoid mixing mieb and mteb tasks
benchmark = mieb.create_benchmark(tasks=tasks)
results = benhcmark.run(model, output_folder=f"results-mieb/{model_name}")
```

Though I would def. still keep a lot of the functionality the same, however during development this simplifies merge conflict and might also be a good way to avoid people running text models on image.

Alternatively:
```suggestion
import mteb

model_name = "openai/clip-vit-large-patch14"
model = mteb.get_model(model_name)

tasks = mteb.get_tasks(tasks=["RenderedSST2"], modalities=["visual"]) # to avoid mixing tasks?
benchmark = mteb.MTEB(tasks=tasks) 
results = benhcmark.run(model, output_folder=f"results-mieb/{model_name}")
```

Reply:
great idea. I will take care of this after tasks are developed!